### PR TITLE
add Automatic-Module-Name to JAR

### LINKED ISSUE #82: Provide a module descriptor for usage with JPMS
completable-futures needs to at least declare an `Automatic-Module-Name` entry for stable use under JPMS.

Ideally, it would provide a full module descriptor, a.k.a a `module-info` file.

### PR DESCRIPTION
**Pull request by mattnworb**
If the JAR for this repo is used in a project where the module system is
active, Java treats any "plain JAR" without a module descriptor as an
"automatic module" and allows it to be used by other modules in the
application.

For instance, a module in an application using JPMS can depend on this
project like

```
module com.mattnworb.jpms {
  requires com.spotify.completablefutures;
}
```

Defining an `Automatic-Module-Name` allows for explicitly controlling
the name that modules use when loading your automatic module, rather
JPMS guessing at the automatic module name for the JAR based on the
JAR's filename.

Alternatively we could add a `module-info.java` to this project to
enable it to be loaded as a regular (not automatic) module but that
would also require increasing the compiler target from 8 to 9.

Closes #82

### CODE REVIEW

**Comment by mattnworb** [comment]
I was curious to learn more about JPMS so when I saw the related issue I thought it would be a good excuse to finally do that :)

File: pom.xml [hunk 1]
```diff
@@ -134,6 +134,17 @@
       <plugin>
         <artifactId>maven-failsafe-plugin</artifactId>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+          <archive>
+            <manifestEntries>
+              <Automatic-Module-Name>com.spotify.completablefutures</Automatic-Module-Name>
```
Comment (mbruggmann) [inline pom.xml:143]: Knowing nothing about JPMS: The package name in use is `com.spotify.futures`. Should they match?

Comment (mattnworb) [reply pom.xml:143]: It seems like that is [the convention](https://blog.joda.org/2017/04/java-se-9-jpms-module-naming.html) but not required, but I figured we'd have a conflict if we ever did the same thing in https://github.com/spotify/futures-extra which has the same package name.

Comment (spkrka) [reply pom.xml:143]: Yes, in futures-extra we have this: https://github.com/spotify/futures-extra/blob/master/src/main/java/module-info.java#L17

**Review by spkrka — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (mattnworb) [reply pom.xml:143]: oh nice, I did not realize there was a proper module-info there already

**Pull request merged**